### ISSUE TITLE
[BUG] pueue 1.0.2 test issue

### ISSUE THREAD

**Issue by chenrui333** [Nukesor/pueue#242]
**Describe the bug**
Error when running `pueue status` for 1.0.2 build

```
$ /usr/local/Cellar/pueue/1.0.2/bin/pueue status
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', client/display/state.rs:97:58
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**To Reproduce**
simply run `pueue status`

**Expected behavior**
show the right result

**Screenshots**
Only needed if the problem is of graphic

**Additional context**
- Operating System macOS
- Pueue version. Can be seen at the top of `pueue -h` output

```
$ pueue -h
Pueue client 1.0.2

Arne Beer <contact@arne.beer>

Interact with the Pueue daemon

USAGE:
    pueue [FLAGS] [OPTIONS] <SUBCOMMAND>

FLAGS:
    -h, --help       Print help information
    -v, --verbose    Verbose mode (-v, -vv, -vvv)
    -V, --version    Print version information

OPTIONS:
    -c, --config <CONFIG>    Path to a specific pueue config file to use. This ignores all other
                             config files

SUBCOMMANDS:
    add            Enqueue a task for execution
    clean          Remove all finished tasks from the list
    completions    Generates shell completion files. This can be ignored during normal
                   operations
    edit           Edit the command or path of a stashed or queued task.
                   The command is edited by default.
    enqueue        Enqueue stashed tasks. They'll be handled normally afterwards
    follow         Follow the output of a currently running task. This command works like tail
                   -f
    group          Use this to add or remove groups. By default, this will simply display all
                   known groups
    help           Print this message or the help of the given subcommand(s)
    kill           Kill specific running tasks or whole task groups. Kills all tasks of the
                   default group when no ids are provided
    log            Display the log output of finished tasks. Prints either all logs or only the
                   logs of specified tasks
    parallel       Set the amount of allowed parallel tasks. By default, adjusts the amount of
                   the default group
    pause          Either pause running tasks or specific groups of tasks.
                   By default, pauses the default group and all its tasks.
                   A paused queue (group) won't start any new tasks.
    remove         Remove tasks from the list. Running or paused tasks need to be killed first
    reset          Kill all tasks, clean up afterwards and reset EVERYTHING!
    restart        Restart task(s). Identical tasks will be created and by default enqueued. By
                   default, a new task will be created
    send           Send something to a task. Useful for sending confirmations such as 'y\n'
    shutdown       Remotely shut down the daemon. Should only be used if the daemon isn't
                   started by a service manager
    start          Resume operation of specific tasks or groups of tasks.
                   By default, this resumes the default group and all its tasks.
                   Can also be used force-start specific tasks.
    stash          Stashed tasks won't be automatically started. You have to enqueue them or
                   start them by hand
    status         Display the current status of all tasks
    switch         Switches the queue position of two commands. Only works on queued and stashed
                   commands
    wait           Wait until tasks are finished. This can be quite useful for scripting. By
                   default, this will wait for all tasks in the default group to finish. Note:
                   This will also wait for all tasks that aren't somehow 'Done'. Includes:
                   [Paused, Stashed, Locked, Queued, ...]
```

---

relates to https://github.com/Homebrew/homebrew-core/pull/85112

**Comment by brunomiguel** [comment]
I'm having the exact same problem

**Comment by adworacz** [comment]
I'm seeing the same thing. 

If it helps, it looks like pueued is blowing up too:
```
● pueued.service - Pueue Daemon - CLI process scheduler and manager
     Loaded: loaded (/etc/xdg/systemd/user/pueued.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2021-09-13 20:51:13 PDT; 2min 19s ago
   Main PID: 2300806 (pueued)
      Tasks: 8 (limit: 9095)
     Memory: 6.3M
        CPU: 419ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/pueued.service
             └─2300806 /usr/bin/pueued

Sep 13 20:53:31 nas pueued[2300806]: 03:53:31 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:31 nas pueued[2300806]: 03:53:31 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:32 nas pueued[2300806]: 03:53:32 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:32 nas pueued[2300806]: 03:53:32 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:32 nas pueued[2300806]: 03:53:32 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:32 nas pueued[2300806]: 03:53:32 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:32 nas pueued[2300806]: 03:53:32 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:33 nas pueued[2300806]: 03:53:33 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:33 nas pueued[2300806]: 03:53:33 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
Sep 13 20:53:33 nas pueued[2300806]: 03:53:33 [ERROR] Got valid group default, but no worker pool has been initialized. This is a bug!
```

**Comment by Nukesor** [comment]
This is a follow up issue due to #236 and changes in v1.0.0

A simple fix for now, would be to manually add the `default` group manually to the configuration.

E.g.
```
  groups:
    default: 1
```

The default group is expected to always be there, but it wasn't before v1.0.0.
Due to the bug above, some settings weren't properly updated, leading to the current bug being hidden until now.
It also didn't pop up during testing between versions, as a lot of breaking changes were introduced i 1.0.0 :/. This made testing basically impossible.

I forgot to think about the possibility, that the `default` group could be absent in the configuration file and therefore expected it to always be present!

This will be an easy fix, but I really don't have much time right now. I guess I'll fix it till saturday.

**Comment by brunomiguel** [comment]
That fixes it for me

**Comment by Nukesor** [comment]
A fix has been released in `pueue-lib` with `v0.18.1` in https://github.com/Nukesor/pueue-lib/commit/b16fe2f05d56e2d3664b27ea75310ae661d5eae5

This follow-up commit https://github.com/Nukesor/pueue/commit/c02286bb5b1ee091186bd3467531e92463297801 introduces that fix to `pueue`. Pueue v1.0.3 has just being released, which will fix this issue.
